botorch/acquisition/acquisition.py

```diff
@@ -37,6 +37,7 @@ def __init__(self, model: Model) -> None:
         """
         super().__init__()
         self.add_module("model", model)
+        self.model: Model = self.model
```

Comment:
What if you just do `self.model = model` here? I believe that if `model` is a `torch.nn.Module` then the module registration should happen automagically under the hood.

Reply:
Yep that works! Making this change and adding tests for proper initialization.